src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.SemanticChangeProcessor.cs

```diff
@@ -185,11 +185,9 @@ private async Task EnqueueWorkItemAsync(Document thisDocument, ImmutableArray<Lo
 
                         var document = solution.GetDocument(location.SourceTree, projectId);
                         if (document == null || thisDocument == document)
-                        {
                             continue;
-                        }
 
-                        await _processor.EnqueueWorkItemAsync(document).ConfigureAwait(false);
+                        await _processor.EnqueueWorkItemAsync(document.Project, document.Id, document).ConfigureAwait(false);
```

Comment:
if we were in a scenario where we actually had the document, then we pass it along to avoid extra work later.

Reply:
we could consdier not passing hte doc with the hope that project.GetDocument(id) is cheap if the red doc was already created.